app/controllers/api/v2/registration_tokens_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module V2
+    class RegistrationTokensController < V2::BaseController
+      include Foreman::Controller::UsersMixin
+
+      include Foreman::Controller::Parameters::User
+      before_action :authenticate, :only => [:invalidate_tokens]
+
+      def resource_class
+        User
+      end
+
+      def find_resource(permission = :view_users)
+        editing_self? ? User.find(User.current.id) : User.authorized(permission).except_hidden.find(params[:id])
+      end
+
+      def action_permission
+        case params[:action]
+        when 'invalidate_tokens', 'destroy'
+          'edit'
+        else
+          super
+        end
+      end
+
+      api :DELETE, '/users/:id/registration_tokens', N_("Invalidate all registration Tokens for a specific user.")
```

Comment:
If a user can only have 0 or 1 registration token, why do we use plural instead of singular phrasing?

Reply:
A user has multiple jwt tokens , but we just store it as one jwt secret. So verbally it makes sense to use plural

Reply:
cc : @[Lennonka](https://github.com/Lennonka)

Reply:
I’d suggest keeping it consistent with the phrasing used in the UI PR.

Reply:
[This](https://github.com/theforeman/foreman/pull/10357/commits/a19ac9110b57bbd60bdaabeb25f1b70f8e65185c#diff-0a32d4e9f83e8c59487817bce4ccece334edc27596e9c7c9e8e2888ed4b0fc97R41) is how the phrasing looks like in the UI PR. They're similar

Reply:
Why do they need to differ at all? Aren't we performing the same action in both cases?